LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -156,17 +247,177 @@ List<Module> GetModuleList()
+                }
+            }
+            else
+            {
+                ReportError(ModuleIdList.ToString(), action);
+            }
+            ReportSummary(action, classroomCount, "classroom");
+            ReportFooter(action);
+
         }
-        
+
         public void AssignInstructorsToClassrooms()
         {
-            throw new NotImplementedException();
+            string action = "Assign instructor to classroom";
+            ReportHeader(action);
+
+            if (ActiveInstructorList.Count > 0 && ClassroomDictionary.Count > 0)
+            {
+                foreach (var classroom in ClassroomDictionary.Values)
+                {
+                    Guid moduleId = classroom.Module.Id;
+                    List<Instructor> instructorList;
+                    if (ModuleInstructorListDictionary.ContainsKey(moduleId))
+                    {
+                        ModuleInstructorListDictionary.TryGetValue(moduleId, out instructorList);
+                        classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
+                    }
+                }
+            }
+            else if (ActiveInstructorList.Count == 0)
+            {
+                ReportError(ActiveInstructorList.ToString(), action);
+                if (ClassroomDictionary.Count == 0)
+                {
+                    ReportError(ClassroomDictionary.ToString(), action);
+                }
+            }
+            ReportSummary(action, 1, "classroom");
+            ReportFooter(action);
+        }
+
+        void AddInstructorToModuleList(Instructor instructor)
+        {
+            List<Module> moduleList = instructor.ModuleList;
+            foreach (var module in moduleList)
+            {
+                List<Instructor> instructorList;
+                if (ModuleInstructorListDictionary.ContainsKey(module.Id))
+                {
+                    ModuleInstructorListDictionary.TryGetValue(module.Id, out instructorList);
+                    instructorList.Add(instructor);
+                }
+                else
+                {
+                    instructorList = new List<Instructor>() { instructor };
+                    ModuleInstructorListDictionary.Add(module.Id, instructorList);
+                }
+            }
         }
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            string action = "Register Students For Classes";
+            int itemListCount = 0;
+            ReportHeader(action);
+
+            if (ClassroomDictionary.Count > 0)
+            {
+                foreach (var classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        itemListCount = classroom.ItemList.Count;
```

Comment:
The logic is incorrect, if you want to report how many student registered for that particular class you should call ReportItem here with this count. But you assign itemListCount in the loop and do not use it there. Instead, you take the last assigned value and use it with the summary, which is wrong.